docker-compose.run-tests.yaml

```diff
@@ -0,0 +1,48 @@
+# Docker compose file for running tests.
+# Note: If testing locally, change the network to external: false.
+
+version: '3.8'
+
+services:
+  db:
+    image: postgres:13.0-alpine
+    restart: always
+    volumes: 
+      - db:/var/lib/postgresql/data
+      - ./server/.env.dev:/docker-entrypoint-initdb.d/.env.dev
+      - ./server/dev-db-init.sh:/docker-entrypoint-initdb.d/dev-db-init.sh
+    environment:
+      - POSTGRES_HOST_AUTH_METHOD=trust  
+    env_file:
+      - ./server/.env.dev
+
+  server:
+    build: ./server
+    image: ${IMAGE_NAME_FOR_RUN}
```

Comment:
Is image name needed if building from a docker file? Since the `${IMAGE_NAME_FOR_RUN}` image is actually built in a later step, and only on main, does naming the image built at this step potentially conflict with that?

Alternatively, if the "build image if main" step was moved to before the test step, and done on all branches, could the docker compose just reference that image instead of building here? Just curious, not suggesting you make that change because I doubt it will work that way post-#78.

Reply:
IMAGE_NAME_FOR_RUN is assigned in make_cloud_build_env_file.sh in the first Cloud Build step, and works because we pass that variable in with --env-file .env.cloud_build in the docker compose up line (which it appears we don't need anymore - because the image name is irrelevant as this is only used for contained testing).  I've removed those flags from the testing cloud build step and hard coded the image name in docker-compose.run-tests.yaml. May need to revisit in the next step to be able to access SHA to tag reports when saving.